internal/commands/app/create_inputs.go

```diff
@@ -153,56 +154,69 @@ func (i *createInputs) resolveLocalPath(ui terminal.UI, wd string) (string, erro
 	return fullPath, nil
 }
 
-func (i *createInputs) resolveCluster(client atlas.Client, groupID string) (dataSourceCluster, error) {
+func (i *createInputs) resolveClusters(client atlas.Client, groupID string) ([]dataSourceCluster, error) {
 	clusters, err := client.Clusters(groupID)
 	if err != nil {
-		return dataSourceCluster{}, err
+		return []dataSourceCluster{}, err
 	}
-	var clusterName string
-	for _, cluster := range clusters {
-		if i.Cluster == cluster.Name {
-			clusterName = cluster.Name
-			break
-		}
+
+	existingClusters := map[string]atlas.Cluster{}
+	for _, c := range clusters {
+		existingClusters[c.Name] = c
 	}
-	if clusterName == "" {
-		return dataSourceCluster{}, errors.New("failed to find Atlas cluster")
+
+	var dsClusters []dataSourceCluster
+	for _, clusterName := range i.Clusters {
+		if _, ok := existingClusters[clusterName]; !ok {
+			continue
+		}
+		dsClusters = append(dsClusters,
+			dataSourceCluster{
+				Name: "mongodb-atlas",
+				Type: "mongodb-atlas",
+				Config: configCluster{
+					ClusterName:         clusterName,
+					ReadPreference:      "primary",
+					WireProtocolEnabled: false,
+				},
+			})
 	}
-	dsCluster := dataSourceCluster{
-		Name: "mongodb-atlas",
-		Type: "mongodb-atlas",
-		Config: configCluster{
-			ClusterName:         clusterName,
-			ReadPreference:      "primary",
-			WireProtocolEnabled: false,
-		},
+	if len(dsClusters) == 0 {
+		return []dataSourceCluster{}, errors.New("failed to find Atlas cluster")
 	}
-	return dsCluster, nil
+	return dsClusters, nil
 }
 
-func (i *createInputs) resolveDataLake(client atlas.Client, groupID string) (dataSourceDataLake, error) {
+func (i *createInputs) resolveDataLakes(client atlas.Client, groupID string) ([]dataSourceDataLake, error) {
 	dataLakes, err := client.DataLakes(groupID)
 	if err != nil {
-		return dataSourceDataLake{}, err
+		return []dataSourceDataLake{}, err
 	}
-	var dataLakeName string
-	for _, dataLake := range dataLakes {
-		if i.DataLake == dataLake.Name {
-			dataLakeName = dataLake.Name
-			break
-		}
+
+	existingDataLakes := map[string]atlas.DataLake{}
+	for _, d := range dataLakes {
+		existingDataLakes[d.Name] = d
 	}
-	if dataLakeName == "" {
-		return dataSourceDataLake{}, errors.New("failed to find Atlas data lake")
+
+	var dsDataLakes []dataSourceDataLake
+	for _, dataLakeName := range i.DataLakes {
+		if _, ok := existingDataLakes[dataLakeName]; !ok {
+			continue
+		}
+
+		dsDataLakes = append(dsDataLakes,
+			dataSourceDataLake{
+				Name: "mongodb-datalake",
+				Type: "datalake",
```

Comment:
same thing about consts, also I missed this in the first implementation but is this really accurate that a datalake would have a different name than type?  (cc: @nathantfrank)

if so, then only the `Type` here needs to be a const

Reply:
Yes this is the case.